Extension/src/main.ts

```diff
@@ -279,6 +287,25 @@ async function postInstall(info: PlatformInformation): Promise<void> {
 }
 
 async function finalizeExtensionActivation(): Promise<void> {
+    if (vscode.workspace.getConfiguration("C_Cpp", null).get<string>("intelliSenseEngine") === "Disabled") {
+        languageServiceDisabled = true;
+        getTemporaryCommandRegistrarInstance().disableLanguageServer();
+        disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
+            if (!reloadMessageShown && vscode.workspace.getConfiguration("C_Cpp", null).get<string>("intelliSenseEngine") !== "Disabled") {
+                reloadMessageShown = true;
+                util.promptForReloadWindowDueToSettingsChange();
+            }
+        }));
+        Telemetry.logLanguageServerEvent("intelliSenseEngine disabled");
+        return;
+    } else {
+        disposables.push(vscode.workspace.onDidChangeConfiguration(() => {
```

Comment:
since the "true" case does an early return, you don't need to put this part in an "else" block.